src/App.tsx

```diff
@@ -1,17 +1,130 @@
-import InputTodo from "./Components/InputTodo";
-import TodoList from "./Components/TodoList";
+import TodoInput from "./Components/TodoInput";
+import TodoItem from "./Components/TodoItem";
+import CreatePortal from "./Components/CreatePortal";
+import { useAppDispatch, useAppSelector } from "./hooks";
+import { selectItem } from "./Features/textItem/item-slice";
+import { IItem } from "./Features/textItem/item-slice";
+/* import { textItemTEST } from "./Features/Item/textItem-slice"; */
+
+import {
+  addItem,
+  /*  editItem, */
+  deleteItem,
+  completedItem,
+} from "./Features/textItem/item-slice";
+
+/* interface ITodos {
+  id: number;
+  todo: string;
+  completed: boolean;
+} */
 
 const App: React.FC = () => {
+  const [todos, setTodos] = useState<IItem[]>([]);
+  const [showEdit, setShowEdit] = useState<boolean>(false);
+  const [edit, setEdit] = useState<string>("");
+
+  const [editId, setEditId] = useState<number>();
+  const [saveEdit, setSaveEdit] = useState<string>("");
+  /*   const [decorationText, setDecorationText] = useState<boolean>(false); */
+  const [todosEdit, setTodosEdit] = useState<string>("");
+  /*   const [textItem, setTextItem] = useState<string>(""); */
+  const dispatch = useAppDispatch();
+  const todo = useAppSelector(selectItem);
+  /*  console.log(todo); */
+
+  useEffect(() => {
+    setTodos(todo);
+  }, [todo]);
+
+  const addTodosHandler = (initialState: IItem) => {
+    console.log(initialState);
+    /*  setTextItem(initialState.todo); */
+    dispatch(addItem(initialState));
+  };
+
+  const editElementHandler = (initialState: IItem) => {
+    console.log("APP-edit");
+    console.log(initialState);
+    setShowEdit((previousState: boolean) => true);
+    setEditId(initialState.id);
+    setTodosEdit(initialState.todo);
+    setEdit(initialState.todo);
+    setSaveEdit(initialState.todo);
+  };
+
+  const deleteElementHandler = (initialState: IItem) => {
+    dispatch(deleteItem(initialState));
+  };
+
+  // МОДАЛЬНОЕ ОКНО
+  //__________________________________________________________
+  const editValueHandler = (textInput: string) => {
+    setEdit(textInput);
+  };
+
+  const cancelShowHandler = () => {
+    setShowEdit((previousState: boolean) => false);
+  };
+
+  const saveShowHandler = () => {
+    console.log("saveShowHandler");
+    console.log(editId);
+    console.log(todosEdit);
+    console.log(todos);
+    console.log(edit);
+
+    /*  dispatch(editItem(edit)); */
+
+    setShowEdit((previousState: boolean) => false);
+  };
+
+  //_________________________________________________________
+
+  const completedElementHandler = (
+    name: string,
+    id: number,
+    checked: boolean
+  ) => {
+    /*  setDecorationText((previousState) => !previousState); */
+    /* console.log(decorationText); */
+
+    let result = todos.find((item) => Number(name) === item.id);
+    if (result) {
+      console.log(result.completed);
+      result.completed = !result.completed;
```

Comment:
never change state values directly, you should always use setState for this. And by changing values here, they don't change in redux store, you should remove local state for todos